deltachat-jsonrpc/src/api/types/chat.rs

```diff
@@ -0,0 +1,91 @@
+use anyhow::{anyhow, Result};
+use deltachat::chat::get_chat_contacts;
+use deltachat::chat::{Chat, ChatId};
+use deltachat::contact::{Contact, ContactId};
+use deltachat::context::Context;
+use num_traits::cast::ToPrimitive;
+use serde::Serialize;
+use typescript_type_def::TypeDef;
+
+use super::color_int_to_hex_string;
+use super::contact::ContactObject;
+
+#[derive(Serialize, TypeDef)]
+pub struct FullChat {
+    id: u32,
+    name: String,
+    is_protected: bool,
+    profile_image: Option<String>, //BLOBS ?
+    archived: bool,
+    // subtitle  - will be moved to frontend because it uses translation functions
+    chat_type: u32,
+    is_unpromoted: bool,
+    is_self_talk: bool,
+    contacts: Vec<ContactObject>,
+    contact_ids: Vec<u32>,
+    color: String,
+    fresh_message_counter: usize,
+    // is_group - please check over chat.type in frontend instead
+    is_contact_request: bool,
+    is_device_chat: bool,
+    self_in_group: bool,
+    is_muted: bool,
+    ephemeral_timer: u32, //TODO look if there are more important properties in newer core versions
+    can_send: bool,
+}
+
+impl FullChat {
+    pub async fn from_dc_chat_id(context: &Context, chat_id: u32) -> Result<Self> {
```

Comment:
`try_from_dc_chat_id` would be more idiomatic, as it is failable

Reply:
what are the rules for this? on every from-function that can fail?
so same for `ContactObject::from_dc_contact`?

Reply:
yes, the rule is try_from for Result returns and from for conversions that always work